### PR TITLE
fix: use correct status codes when resources are missing

### PR DESCRIPTION
**Pull request by water-sucks**
There were quite a lot of misleading status codes, especially
with the flight plans. This kind of sucks, since the frontend
usually relies on these status codes for the existence of resources.

Additionally, some responses weren't being parsed correctly due to
the JSON encoder just refusing to deserialize zero-length slices properly.
Welp. I fixed most of those as well.

I also added some consistency guarantees; flight plans with a single plan
cannot become empty now, and deletion happens much more idempotently given
resources exist.

### CODE REVIEW

File: database/query/flight_plan.sql
Comment (MattRStoffel) [inline database/query/flight_plan.sql:1]: This doesn't seem related to status codes

**Review by MattRStoffel — APPROVED** [review]
Those status codes look much better than what we had before. Code quality is up to par. Everything looks good to me.

File: database/query/flight_plan.sql
Comment (water-sucks) [reply database/query/flight_plan.sql:1]: Funny enough, in order to bring up the correct status codes, I needed to hit the database a few more times with some explicit queries for checking existence.

Go usually uses zero values to indicate no data, which makes checking for real existence inside a database a little more complex; I offsetted this to the database because of it for ease of implementation.
